examples/hitl/rearrange_v2/rearrange_v2.py

```diff
@@ -83,15 +91,29 @@ def get_objects_state(self):
             )
         return object_states
 
-    def record_state(self, task_completed: bool = False):
-        agent_states = self.get_agents_state()
-        object_states = self.get_objects_state()
-
-        self._app_service.step_recorder.record("agent_states", agent_states)
-        self._app_service.step_recorder.record("object_states", object_states)
-        self._app_service.step_recorder.record(
-            "task_completed", task_completed
-        )
+    def record_state(
+        self, elapsed_time: float, user_data: List[UserData]
+    ) -> Dict[str, Any]:
+        data: Dict[str, Any] = {
+            "t": elapsed_time,
+            "users": [],
+            "object_states": self.get_objects_state(),
+            "agent_states": self.get_agents_state(),
+        }
+
+        for user_index in range(len(user_data)):
+            u = user_data[user_index]
+            user_data_dict = {
+                "task_completed": u.episode_finished,
+                "task_succeeded": u.episode_success,
+                "camera_transform": u.cam_transform,
+                "held_object": u.ui._held_object_id,
```

Comment:
We've discussed this already but this ID if passed to grasp_mgr, in a way understood by sim, would solve what habitat-llm needs.